addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -186,14 +365,82 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
     /**
      * @see this.selectClass for params
      */
-    style: function (previewMode, widgetValue, params) {
-        this._rpc({
-            route: '/shop/change_styles',
-            params: {
-                'id': this.productTemplateID,
-                'style_id': params.possibleValues[params.possibleValues.length - 1],
-            },
+    async setRibbon(previewMode, widgetValue, params) {
+        this.$target[0].dataset.ribbonId = widgetValue;
+        this.trigger_up('set_product_ribbon', {
+            templateId: this.productTemplateID,
+            ribbonId: widgetValue || false,
         });
+        const ribbon = this.ribbons[widgetValue] || {name: '', color: '', text_color: '', html_class: ''};
+        const $ribbons = $(`[data-ribbon-id="${widgetValue}"] .o_ribbon:not(.o_wsale_ribbon_dummy)`);
+        $ribbons.text(ribbon.name);
+        $ribbons.attr('class', 'o_ribbon');
+
+        if (!ribbon.html_class) {
+            $ribbons.css('background-color', ribbon.color);
+        } else {
+            $ribbons.addClass(ribbon.html_class);
+            $ribbons.css('background-color', '');
+        }
```

Comment:
```javascript
$ribbons.css('background-color', ribbon.color ? ribbon.color : '');
if (ribbon.html_class) {
     $ribbons.addClass(ribbon.html_class);
}
```
would that be ok ? Would be more generic and allow both a class and a color as it may be the case in a future implementation I would like to do.